src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -827,6 +829,13 @@ private async Task<T> ExecuteWithFailOverPolicyAsync<T>(IEnumerable<Configuratio
                     T result = await funcToExecute(currentClient).ConfigureAwait(false);
                     success = true;
 
+                    Uri currentEndpoint = _configClientManager.GetEndpointForClient(currentClient);
+
+                    if (failureOccurred && originalEndpoint != currentEndpoint)
```

Comment:
This would log only when (if) there is a successful operation. We should be logging all endpoints in the same order that they were tried in.